test_pull_requests

```diff
@@ -152,10 +159,50 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+# We do not expect the groups used to define trusted and administrative
+# members for this repo to change often, so we will 'cache' the group ID
+# to GitHub name and URL for them in a file on disk.
+TEAM_CACHE_LOCATION = File.expand_path("~/.team_cache.json")
+$team_cache = if File.exists? TEAM_CACHE_LOCATION
+  JSON.parse(IO.read(TEAM_CACHE_LOCATION))
+else
+  {}
+end
+
 # Dynamically extend GitHubAPI to add methods that don't
 # require a local git project
 module Hub
   class GitHubAPI
+
+    # team_url_and_name returns the user-friendly URL and name for a GitHub team.
+    # If the team ID is found in the $team_cache, we will simply use it. If it
+    # is not, however, we will need to expend an API request to list all of the
+    # teams we care about and fill in the $team_cache before continuing.
+    def team_url_and_name(team_id)
+      # We want to return our cached data whenever we have it, but we also want
+      # to refresh our cache once in a while, so with a 1% chance we will ignore
```

Comment:
The rest of the code doesn't have a file cache.  So this is different.  But I don't feel that strongly either.